src/components/page-header/page-header.component.html

```diff
@@ -85,6 +85,7 @@ <h1 class="page-header-title">{{header}}</h1>
         <ux-tabset *ngIf="(selectedRoot$ | async)?.children; let children" [manual]="true">
             <ux-tab *ngFor="let child of children"
                 [heading]="child.title"
+                [routerPath]="child.routerLink"
                 [active]="child === (selected$ | async)"
                 (activated)="select(child)"
```

Comment:
I'm a bit concerned that this is going to cause multiple route events, because the select function calls `navigate`.

https://github.com/UXAspects/UXAspects/blob/master/src/components/page-header/page-header.service.ts#L44

Since you've applied the routerLink directive, this call to `navigate` should no longer be called on a user click/keypress. However, if the tab is activated via the autoselect behavior, it should still navigate.

Reply:
@ashley-hunter I'd like to get your opinion on this if you get the chance.

Reply:
Yeah this is a good point. 

I was thinking perhaps we could check if the routerLink is active and then skip navigation if that was the case, but I don't even think that would work as there are cases like lazy loaded routes and route guards which are asynchronous so navigation may not have occurred by the time we check.

Perhaps we could add an optional parameter to the select function to skip navigation?

E.g.

```typescript
select(item: PageHeaderNavigationItem, navigate: boolean = true): void {
...
    if (item.routerLink && navigate) {

            // Trigger router navigation
            const routerLink = Array.isArray(item.routerLink) ? item.routerLink : [item.routerLink];
            this._router.navigate(routerLink, item.routerExtras);

        }
```

```html
(activated)="select(child, false)"
```

Or if there may not always be a routerLink then:
```html
(activated)="select(child, !!child.routerLink)"
```

Reply:
Yes, I think `select(child, false)` should work. The `activated` event is also emitted when the tab changes programmatically, but I think that only happens with the `secondaryNavigationAutoselect` option and there's code to cover that (as far as I can tell).